packages/ui/Dropdown.tsx

```diff
@@ -12,21 +12,24 @@ export const DropdownMenuTrigger = forwardRef<HTMLButtonElement, DropdownMenuTri
       className={
         props.asChild
           ? className
-          : `relative inline-flex items-center rounded-sm bg-transparent px-3 py-2 text-sm font-medium text-gray-700 hover:bg-gray-50 focus:bg-gray-100 focus:outline-none focus:ring-2 focus:ring-neutral-500 focus:ring-offset-1 group-hover:text-black ltr:ml-2 rtl:mr-2 ${className}`
+          : `inline-flex items-center rounded-sm bg-transparent px-3 py-2 text-sm font-medium text-gray-700 hover:bg-gray-50 focus:bg-gray-100 focus:outline-none focus:ring-2 focus:ring-neutral-500 focus:ring-offset-1 group-hover:text-black ${className}`
       }
       ref={forwardedRef}
     />
   )
 );
 DropdownMenuTrigger.displayName = "DropdownMenuTrigger";
 
+export const DropdownMenuTriggerItem = DropdownMenuPrimitive.TriggerItem;
+
 type DropdownMenuContentProps = ComponentProps<typeof DropdownMenuPrimitive["Content"]>;
 export const DropdownMenuContent = forwardRef<HTMLDivElement, DropdownMenuContentProps>(
   ({ children, ...props }, forwardedRef) => {
     return (
       <DropdownMenuPrimitive.Content
         {...props}
-        className="z-10 mt-1 w-48 origin-top-right rounded-sm bg-white text-sm shadow-lg ring-1 ring-black ring-opacity-5 focus:outline-none"
+        portalled={false}
```

Comment:
This change is very important. This makes the Dropdown not added under `body` by default which was messing around with the Dialog which also gets added by default under `body`. Having the two in the same context with no way to modify their out-of-the-box z-index makes the dialog to be seen above the dialog in the overlay.